packages/web3/test/integration/web3.config.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { SupportedProviders, Web3BaseProvider } from 'web3-types';
+import {
+	closeOpenConnection,
+	createTempAccount,
+	getSystemTestProvider,
+	isWs,
+	waitForOpenConnection,
+} from '../shared_fixtures/system_tests_utils';
+import Web3 from '../../src/index';
+// import { BasicAbi, BasicBytecode } from '../shared_fixtures/build/Basic';
+
+describe('Web3 instance', () => {
+	let provider: string | SupportedProviders;
+	let accounts: string[];
+	let web3: Web3;
+	let currentAttempt = 0;
+
+    beforeEach(() => {
+		currentAttempt = 0;
+	});
+
+	beforeAll(async () => {
+		provider = getSystemTestProvider();
+		const acc1 = await createTempAccount();
+		const acc2 = await createTempAccount();
+		accounts = [acc1.address, acc2.address];
+		web3 = new Web3(provider);
+	});
+	afterAll(async () => {
+		await closeOpenConnection(web3);
+	});
+
+    afterEach(async () => {
+		if (isWs) {
+			// make sure we try to close the connection after it is established
+			if (
+				web3?.provider &&
+				(web3.provider as unknown as Web3BaseProvider).getStatus() === 'connecting'
+			) {
+				await waitForOpenConnection(web3, currentAttempt);
+			}
+
+			if (web3?.provider) {
+				(web3.provider as unknown as Web3BaseProvider).disconnect(1000, '');
+			}
+		}
+	});
+
+    it('should be send transaction, change for defaultTransactionType and sucesfully send transaction with different type', async () => {
```

Comment:
also add tests for other imp configs like `contractDataInputFill` at web3 level here for all possible values,

Reply:
i agree, will create another issue because we have alot of configs and will be time consuming